### PR TITLE
Add mini herbie tutorial to our tests

### PR DESCRIPTION
**Pull request by oflatt**

### CODE REVIEW

File: tests/herbie-tutorial.egg [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+(let target
+  (Add 
+    (Add (Num (rational 1 100)) (Num (rational 1 100)))
+    (Num (rational -2 100))))
+
+(run 1)
+
+;; error is bad, constant folding hasn't fired enough
+(extract (best-error target))
```
[the remark this answers is not in the document]
Comment (oflatt) [reply tests/herbie-tutorial.egg:129]: Great question!
So in practice what I do is I mirror the data type. So I have an ASTMath which I keep separate, and never call union on. Then I have another table which keeps track of the best ASTMath which was used to compute the most accurate number.

Actually, this problem will hopefully be solved soon in egglog by decoupling the idea of a data type from the equality relation.

[the remark this answers is not in the document]
Comment (saulshanabrook) [reply tests/herbie-tutorial.egg:129]: Oh interesting! Do you do that in the larger `herbie.egg` example, or is that table out of band from egglog?

> Actually, this problem will hopefully be solved soon in egglog by decoupling the idea of a data type from the equality relation.

Cool, I would love to hear more about this.

[the remark this answers is not in the document]
Comment (oflatt) [reply tests/herbie-tutorial.egg:129]: The `herbie.egg` example is very out of date, and far from what my experimental version of Herbie actually does. It generates egglog code here:
https://github.com/herbie-fp/herbie/blob/oflatt-egglog-local-area/src/egglog/run-egglog.rkt

[the remark this answers is not in the document]
Comment (oflatt) [reply tests/herbie-tutorial.egg:129]: As for the terms, keep an eye out for a PR. I'm thinking of doing a blog post as well

**Pull request merged**